yarn-project/acir-simulator/src/client/private_execution.test.ts

```diff
@@ -171,9 +166,17 @@ describe('Private Execution test suite', () => {
     let currentNoteIndex = 0n;
 
     const buildNote = (amount: bigint, owner: AztecAddress, storageSlot = Fr.random()) => {
+      // WARNING: this is not actually how nonces are computed!
```

Comment:
Please can you elaborate on the comment, explaining why it's ok for the nonce to not be computed correctly in this test?

Reply:
```suggestion
      // WARNING: this is not actually how nonces are computed!
      // For the purpose of this test we just use the note index here which is fine because
      // nonces are only enforced later by the kernel/later circuits which are not relevant to this test.
      // In practice, the kernel first squashes all transient noteHashes with their matching nullifiers.
      // It then reorders the remaining "persistable" noteHashes. The indexes in that final ordering
      // are used to derive nonce via: `hash(first_nullifier, noteHashIndex)`
```